ospfd/ospf_opaque.h

```diff
@@ -77,6 +77,46 @@
 	((ntohs((lsahdr)->length) >= sizeof(struct lsa_header))                \
 	 && ((ntohs((lsahdr)->length) % sizeof(u_int32_t)) == 0))
 
+/*
+ * Following section defines generic TLV (type, length, value) macros,
+ * used for various LSA opaque usage e.g. Traffic Engineering.
+ */
+struct tlv_header {
+  u_int16_t     type;                   /* Type of Value */
```

Comment:
Using this script will give you an abundance of warnings/errors that need to be cleaned up.

https://github.com/torvalds/linux/blob/master/scripts/checkpatch.pl